.github/workflows/post-promote-production.yml

```diff
@@ -0,0 +1,154 @@
+name: post-promote-production
+
+on:
+  repository_dispatch:
+    types:
+      - post-promote-production
+      - post-promote-production::*
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+
+jobs:
+  # The CRT workflow that calls this workflow has a timeout of 100 minutes, at which point it uses the API
+  # to cancel the workflow, so using an infinite loop rather than a specific number of retries.
+  registry-check:
+    name: Check Terraform Registry for Latest Version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.release-data.outputs.version }}
+    steps:
+      - name: Await Ingest
+        id: release-data
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          LATEST_GITHUB_RELEASE=$(gh release list --repo hashicorp/terraform-provider-aws --json name,isLatest --jq '.[] | select(.isLatest).name')
+
+          while :
+          do
+            LATEST_REGISTRY_DATA=$(curl -s "https://registry.terraform.io/v2/providers/323/provider-versions/latest")
+            LATEST_REGISTRY_VERSION=$(echo "$LATEST_REGISTRY_DATA" | jq -r '.data.attributes.tag')
+
+            if [[ "$LATEST_GITHUB_RELEASE" == "$LATEST_REGISTRY_VERSION" ]]; then
+              echo "Registry is now in sync with GitHub"
+              echo "version=$(echo $LATEST_REGISTRY_DATA | jq -r '.data.attributes.version')" >> "$GITHUB_OUTPUT"
+              break
+            else
+              echo "Latest Registry version ($LATEST_REGISTRY_VERSION) does not yet match lastest GitHub version ($LATEST_GITHUB_RELEASE). Waiting..."
+              sleep 300
+            fi
+          done
+
+  init-check:
+    name: Initialize Provider
+    needs: [registry-check]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: hashicorp/setup-terraform@b9cd54a3c349d3f38e8881555d616ced269862dd
+        with:
+          terraform_wrapper: false
+
+      - name: Set Provider Version in Terraform Configuration
+        run: |
+          cat <<EOF > main.tf
+          terraform {
+            required_providers {
+              aws = {
+                source  = "hashicorp/aws"
```

Comment:
Should this be awscc ?

Reply:
Great catch!

Reply:
The provider id in line 34 might be `2242` for awscc and the `LATEST_GITHUB_RELEASE` var is referring the aws provider as well..

Reply:
lol I was _just_ looking for that number when correcting the other issue. Thanks for finding it for me! 🙂